src/main/java/lotto/domain/Money.java

```diff
@@ -0,0 +1,53 @@
+package lotto.domain;
+
+import lotto.view.input.error.InputError;
+import lotto.view.input.error.InputIllegalArgumentException;
+
+public class Money {
+
+    private static final int ESSENTIAL_RESULT = 0;
+    private static final int LOTTERY_PRICE = 1000;
+
+    private final long money;
+    private final long lotteryCount;
+
+    private static Money moneyInstance;
+
+    private Money(long money, long lotteryCount) {
+        this.money = money;
+        this.lotteryCount = lotteryCount;
+    }
+
+    public static Money getInstance(long amount) {
+        if (moneyInstance == null) {
+            validateMoney(amount);
+            return new Money(amount, amount / LOTTERY_PRICE);
+        }
+        return moneyInstance;
+    }
```

Comment:
money가 싱글톤 방식으로 생성되는 이유가 궁금합니다!
단순하게 생각헸을 때 이렇게 생성자로 바로 받아도 되지 않을까해서요!

```suggestion
    private Money(long money) {
        validateMoney(money);
        this.money = money;
        this.lotteryCount = amount / LOTTERY_PRICE;
    }
```

Reply:
사실 특정 이유가 있다기 보다는 MD 파일에 적은 것처럼
프리코스 동안  최대한 많은 것을 경험해보고 싶었어요!

그래서 싱글톤 방식의 경우에도 위와 같이 사용해본 것 인데요!
현재는 사실 인스턴스를 중복해서 사용하지도, 많이 공유하지도 않기 때문에 현지님이 말씀하신대로 생성자로 받아도 문제가 없다고 생각해요!! 
늘 좋은 의견 감사해요!! 🥰